### PR TITLE
Initial sketch of ATR class

### LINKED ISSUE #19: COD prototype: initial ATR class

### PR DESCRIPTION
**Pull request by ajfabbri**
Fixes #19

### CODE REVIEW

**Comment by ajfabbri** [comment]
FYI: @rutheferd  I forgot to run style formatting on this before pushing. You can do a `npm run format` then `git commit amend` and `git push -f` to fixup that first commit and clear the CI issues.

**Comment by rutheferd** [comment]
@ajfabbri Are we assuming that the autonomy class will capture images using the camera class? Or will the ATR use the camera to capture images in the detection loop. I am thinking that the detection loop is using the camera class to capture images at some rate, and then saving detections (as detectedObject) to the list. Maybe I am going too far into the sketch though?

**Comment by ajfabbri** [comment]
Yep. Once I call atr.start() you periodically snap a pic and add any positive ids to its internal list. Autonomy consumes them one at a time when it is ready.

**Comment by kitplummer** [comment]
We should fix the duplication GHA happening on both pull_request and push actions.

**Comment by rutheferd** [comment]
@ajfabbri I have a rudimentary implementation now. The poll detection should now take the first object in the queue produced by the model prediction random hit stub. Once a hit, then currently empty data is added to the object and it is appended.

**Comment by kitplummer** [comment]
> We should fix the duplication GHA happening on both pull_request and push actions.

Looks like this is still happening.  Wasted GHA time, so please fix.

**Comment by rutheferd** [comment]
> > We should fix the duplication GHA happening on both pull_request and push actions.
> 
> Looks like this is still happening. Wasted GHA time, so please fix.

Alright, given how @ajfabbri and I are working I went ahead and removed the push even all together. Aaron feel free to change if you think that is wrong, did some digging and seems to make sense given that we are beginning work with our PRs. We might want to can "main" to "*" if we expect to have nested branches though!

File: src/atr.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+export class BoxCoordinates {
+    x: number
+    y: number
+
+    constructor(x: number, y: number) {
+        this.x = x
+        this.y = y
+    }
+
+    async getBoxArray(): Promise<Array<number>> {
```
Comment (ajfabbri) [inline src/atr.ts:13]: nit (can fix later): doesn't need to by async. Note: sometimes I'll mark a trivial (never-blocking) function like this as `async` even if it doesn't need to be, when I plan on later adding more logic to it that may require blocking.  In this case I'd expect it to remain a trivial getter.

**Review by ajfabbri — COMMENTED** [review]
Nice, LGTM. Can you please squish (fixup) commits that fix format/lint of a previous commit into the same? Not a huge deal but good habit to get into.

File: src/atr.ts (hunk 1 above)
Comment (rutheferd) [reply src/atr.ts]: Fixed this up! Also went ahead and cleaned up linting commits.

**Pull request merged**